### ISSUE TITLE
Current time should be passable

### ISSUE THREAD

**Issue by magwo** [auth0/node-jsonwebtoken#240]
I suggest allowing options objects to contain current time when verifying and signing.

It's pretty hard to test implementations/integrations involving exp, nbf and timestamps when the library is doing `Date.now()` in various places.

This is even evident in the own tests of this library, with the presence of `fakeDate.js`, which is a hacky global monkey-patching solution to a classic testability problem.
Having, instead, the option to pass current time to verify/sign functions would make things much more straightforward, both for internal tests and the users of the library wishing to write tests.

Example (using lodash for clarity):

```
  if(!_.isDate(options.now)) {
    options.now = Date.now();
  }
  ...
  if (Math.floor(options.now / 1000) >= payload.exp + (options.clockTolerance || 0)) {
    ...
```

**Comment by pjerem** [comment]
> Generated jwts will include an iat (issued at) claim by default unless noTimestamp is specified. If iat is inserted in the payload, it will be used instead of the real timestamp for calculating other things like exp given a timespan in options.expiresIn.

From what I understand, you can :)

**Comment by magwo** [comment]
Hmm yeah actually, the sign functionality supports it.

But the verify stuff still has testability problems regarding current time. Things would be simpler if current time could be passed into verify.

**Comment by gerges-beshay** [comment]
What about adding a flag to use UTC for these checks? I was actually planning on working on that. Currently I'm disabling the time checks and doing it manually using UTC after it passes the other checks.

**Comment by ziluvatar** [comment]
Closed via #274 
You can add `clockTimestamp` option.
